## Hypothesis Testing Assignment
### Customer Order Form.csv

Here we have to check whether defective % varies by center or not.
Here,

Ho = customer order defective % does not vary by centre

Ha = customer order defective % varies by centre

In [1]:
#Importing the libraries
import pandas as pd 
import numpy as np 
from scipy import stats
from scipy.stats import chi2_contingency

In [2]:
df = pd.read_csv(r'C:\Users\91901\Desktop\Costomer+OrderForm.csv')
df.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [3]:
df.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [4]:
df.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [5]:
df.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [6]:
df.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [7]:
#making a list of arrays from individual center values
data = np.array([[271,267,269,280],[29,33,31,20]])
data

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [8]:
#Chi2 contingency test
chi2_contingency(data)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

Here from above chi2 test we can see that;

Chi2 statistic value is 3.858960685820355;

P value is  0.277102099123313.

As P value=0.277102099123313 > 0.05
we accept the null hypothesis.
thus, customer order defective % does not vary by center.